pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -1331,3 +1351,28 @@ func TestDeploymentConfigChanged(t *testing.T) {
 		}
 	}
 }
+
+func TestDurationToHAProxyTimespec(t *testing.T) {
+	testCases := []struct {
+		inputDuration  time.Duration
+		expectedOutput string
+	}{
+		{0, "0s"},
+		{100 * time.Millisecond, "100ms"},
+		{1500 * time.Millisecond, "1500ms"},
+		{10 * time.Second, "10s"},
+		{60 * time.Second, "1m"},
+		{90 * time.Second, "90s"},
+		{2 * time.Minute, "2m"},
+		{75 * time.Minute, "75m"},
+		{120 * time.Minute, "2h"},
+		{48 * time.Hour, "48h"},
+		{720 * time.Hour, "2147483647ms"},
+	}
+	for _, tc := range testCases {
+		output := durationToHAProxyTimespec(tc.inputDuration)
+		if output != tc.expectedOutput {
+			t.Fatalf("Expected \"%s\", got \"%s\"", tc.expectedOutput, output)
```

Comment:
It'd make sense to use `t.Errorf` and `%q`, but no need to hold up this PR for that.  
```suggestion
			t.Errorf("Expected %q, got %q", tc.expectedOutput, output)
```

Reply:
Noted. If CI goes particularly bad I'll make this edit, but otherwise I'll try remember it for a future change.

Reply:
Included this change in the required rebase